pylxd/tests/testing.py

```diff
@@ -35,3 +35,6 @@ def tearDown(self):
     def add_rule(self, rule):
         """Add a rule to the mock LXD service."""
         mock_services.update_http_rules([rule])
+
```

Comment:
This `is_mocked()` isn't required as the `def setUp()` above calls `mock_services.start_http_mock()` which means that when the test runs and calls `is_mocked()` it would always be true.  No other test seems to need it?

Reply:
I'm adding `is_mocked` so that the test suite could also be run against a real LXD. Is that not a goal?

Reply:
Nope, there's no need to run the tests against a real LXD.  These are unit tests as opposed to integration tests.  There are some integration tests, but we probably need a better story on it.  That's going to be one of the goals going forward; it is used (partially) in nova-lxd (part of OpenStack) and gets some integration tests there, but doesn't have anything like a full integration test.

Equally, it is just converting 'python calls' into REST-like API calls; and these are checked in the mocks.  So that's not too bad.

In summary, no the `is_mocked()` isn't needed.

Reply:
OK, but you don't see a problem with me mangling the mock to change the api_extensions as I am doing?